jaxrs/src/main/java/io/airlift/jaxrs/tracing/TracingFilter.java

```diff
@@ -0,0 +1,149 @@
+package io.airlift.jaxrs.tracing;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import org.glassfish.jersey.server.ContainerRequest;
+
+import javax.annotation.Priority;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.HttpHeaders;
+
+import java.net.URI;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.base.Strings.nullToEmpty;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+@Priority(0)
+public final class TracingFilter
+        implements ContainerRequestFilter, ContainerResponseFilter
+{
+    static final String REMOTE_ADDRESS = "airlift.remote-address";
+    static final String REQUEST_SCOPE = "airlift.trace-scope";
+    static final String REQUEST_SPAN = "airlift.trace-span";
+
+    private final TextMapPropagator propagator;
+    private final Tracer tracer;
+    private final String className;
+    private final String methodName;
+
+    public TracingFilter(OpenTelemetry openTelemetry, Tracer tracer, String className, String methodName)
+    {
+        this.propagator = openTelemetry.getPropagators().getTextMapPropagator();
+        this.tracer = requireNonNull(tracer, "tracer is null");
+        this.className = requireNonNull(className, "className is null");
+        this.methodName = requireNonNull(methodName, "methodName is null");
+    }
+
+    @Override
+    public void filter(ContainerRequestContext requestContext)
+    {
+        Context parent = propagator.extract(Context.root(), requestContext, JaxrsTextMapGetter.INSTANCE);
+
+        ContainerRequest request = (ContainerRequest) requestContext.getRequest();
+
+        String method = request.getMethod().toUpperCase(ENGLISH);
+        String remoteAddress = (String) request.getProperty(REMOTE_ADDRESS);
```

Comment:
consider moving this down to the section with `userAgent` and making it conditional.  The remote address could be there, but might not